pkg/cloudprovider/vsphere/instances.go

```diff
@@ -171,5 +171,27 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (i *instances) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
 	klog.V(4).Info("instances.InstanceShutdownByProviderID() called")
-	return false, cloudprovider.NotImplemented
+
+	// Check if node has been discovered already
+	uid := GetUUIDFromProviderID(providerID)
+	if _, ok := i.nodeManager.nodeUUIDMap[uid]; !ok {
+		// IF the uuid is not cached, we end up here
+		klog.V(2).Info("instances.InstanceShutdownByProviderID() NOT CACHED")
+		if err := i.nodeManager.DiscoverNode(uid, cm.FindVMByUUID); err != nil {
+			klog.V(4).Info("instances.InstanceShutdownByProviderID() NOT FOUND with ", uid)
+			// if we can't discover, return false with an error in tow
+			return false, err
+		}
+		klog.V(2).Infof("instances.InstanceShutdownByProviderID() EXISTS with %q", uid)
+	}
+
+	vm, err := i.nodeManager.shakeOutNodeIDLookup(ctx, uid, cm.FindVMByUUID)
+	if err != nil {
+		klog.V(2).Infof("instances.InstanceShutdownByProviderID() shakeOutNodeIDLookup failed with %q", err)
+		return false, err
+	}
+	active, err := vm.VM.IsActive(ctx)
```

Comment:
Just after the call to `IsActive()` can we add a log statement like below in order to observe the result:
```
klog.Infof("VM=%s IsActive=%t", uid, active)
```

Reply:
done 👍